File: leetcode/3-longest-substring-without-repeating-characters/python/longest_substring_without_repeating_characters_brute_force.py
class Solution(object):
    def length_of_longest_substring(self, s: str) -> int:
        n = len(s)
        res = 0
        for i in range(n):
            for j in range(i, n):
                if self.check_repetition(s, i, j):
                    res = max(res, j - i + 1) 
        return res

    def check_repetition(self, s: str, start: int, end: int) -> bool:
        chars = set()
        for i in range(start, end + 1):
            c = s[i]
            if c in chars:
                return False
            chars.add(c)
        return True


sol = Solution()
s = 'abcabcbb'
print(s)
print(sol.length_of_longest_substring(s))

s = 'bbbbb'
print(s)
print(sol.length_of_longest_substring(s))

s = 'pwwkew'
print(s)
print(sol.length_of_longest_substring(s))

File: leetcode/3-longest-substring-without-repeating-characters/javascript/longest-substring-without-repeating-characters-brute-force.js
let lengthOfLongestSubstring = function(s) {
    let n = s.length;
    let res = 0;
    for (let i = 0; i < n; i += 1) {
        for (let j = i; j < n; j += 1) {
            if (checkRepetition(s, i, j)) {
                res = Math.max(res, j - i + 1)
            }
        }
    }
    return res;
};

let checkRepetition = function(s, start, end) {
    const set = new Set();
    for (let i = start; i <= end; i += 1) {
        const c = s[i];
        if(set.has(c)) {
            return false;
        }
        set.add(c);
    }
    return true;
}

let s = 'abcabcbb';
console.log(s);
console.log(lengthOfLongestSubstring(s));

s = 'bbbbb';
console.log(s);
console.log(lengthOfLongestSubstring(s));

s = 'pwwkew';
console.log(s);
console.log(lengthOfLongestSubstring(s));